synapse/cores/common.py

```diff
@@ -457,12 +457,23 @@ def _addDataModels(self, modtups):
 
             for fnam, fnfo, props in modl.get('forms', ()):
 
-                self.addRuntNode('syn:form', fnam, **fnfo)
-                self.addRuntNode('syn:prop', fnam, **fnfo)
+                dmpnfo = self.getPropDef(fnam)[1]
+                dmfnfo = dmpnfo.copy()
+                [dmfnfo.pop(key) for key, valu in list(dmpnfo.items()) if valu is None]
+                self._normTufoProps('syn:form', dmfnfo)
+                self.addRuntNode('syn:form', fnam, **dmfnfo)
+                dmfnfo = dmpnfo.copy()
+                [dmfnfo.pop(key) for key, valu in list(dmfnfo.items()) if valu is None]
+                self._normTufoProps('syn:prop', dmfnfo)
+                self.addRuntNode('syn:prop', fnam, **dmfnfo)
 
                 for pnam, pnfo in props:
                     full = fnam + ':' + pnam
-                    self.addRuntNode('syn:prop', full, **pnfo)
+                    dmpnfo = self.getPropDef(full)[1]
+                    dmpnfo = dmpnfo.copy()
+                    [dmpnfo.pop(key) for key, valu in list(dmpnfo.items()) if valu is None]
+                    self._normTufoProps('syn:prop', dmpnfo)
+                    self.addRuntNode('syn:prop', full, **dmpnfo)
```

Comment:
Calls with kwarg expansion create a new dictionary instance.  any reason the above block couldn't be
```
pdef = self.getPropDef(full)
self.addRuntNode('syn:prop', full, **pdef[1])
```